test/server.js

```diff
@@ -28,6 +28,10 @@ var routes = {
     });
     res.end('hi');
   },
+  '/nonascii': function(res) {
```

Comment:
Maybe just return some binary bytes from this test endpoint? Maybe even something thats invalid utf-8 just for good coverage.